docs/page-objects/input-validation.rst

```diff
@@ -0,0 +1,95 @@
+.. _input-validation:
+
+================
+Input validation
+================
+
+Sometimes the data that your page object receives as input may be invalid.
+
+You can define a ``validate_input`` method in a page object class to check its
```

Comment:
I think it could be worth mentioning that `validate_input()` is called within `.to_item()` method. This reminds users who completely override the `.to_item()` method without calling super to also call `validate_input()` .

Reply:
There also might be some examples in the docs to update, where we define to_item method.

Reply:
There could also be cases where you don't want to call super() in to_item, but still want to have the validation.

Reply:
I have addressed comments 1 and 3.

@kmike I am unsure about modifying existing examples to include `super().to_item` or input validation code, as it would complicate those examples. I think covering how to apply validation from a custom `to_item` as you suggested later may be enough.

Reply:
@kmike Sorry to get back to you about this, after we just discussed it a few hours ago off-GitHub, but could you elaborate on which examples you would update and how? Maybe put 1 example?

Reply:
See https://github.com/scrapinghub/web-poet/pull/113/files#r1143860051. I just realized that `super().to_item()` may not be the right suggestion.